src/components/Wowerlay.tsx

```diff
@@ -103,69 +123,97 @@ export const Wowerlay = defineComponent({
       window.removeEventListener('click', handleWindowClick);
     });
 
+    /**
+     * This variable is for watching Popover's animation state.
+     * When we unmount popover if user passed a transition we need to wait
+     * for transition to end to hide Popover.
+     *
+     * When popover animation ends this variable will hide background.
+     */
     const backgroundVisible = ref(props.visible);
 
+    /**
+     * Hides background element (so popover) when Popover animation ends.
+     */
+    const handleContentTransitionEnd = () => {
+      backgroundVisible.value = false;
+    };
+
     watch(
       () => props.visible,
       (state) => {
         if (state) {
+          // We need to prevent closing popover in the same event loop because
+          // If users clicks a button and changes visible to true
+          // Wowerlay will be opened but same click event may be triggered on window
+          // due to bubbling so it will be cached by our Window:click listener.
           setTimeout(() => {
-            tooltipIsClosable.value = true;
+            popoverClosable.value = true;
           }, 0);
 
           backgroundVisible.value = true;
         } else {
-          tooltipIsClosable.value = false;
-          if (props.transition === false) {
-            setTimeout(() => {
-              backgroundVisible.value = false;
-            });
-          }
+          popoverClosable.value = false;
         }
       },
     );
 
-    const handleContentTransitionEnd = () =>
-      setTimeout(() => {
-        backgroundVisible.value = false;
+    const handleTransition = (type: 'enter' | 'leave', el: Element, done: () => void) => {
+      // We need to wait for FloatingUI to calculate Popover position
+      // If we run this function immediately, it will run before FloatingUI calculations
```

Comment:
Your second line offers some insight, however, it might be somewhat redundant as its implication could be inferred from our first line.